### PR TITLE
Adds EnumInjector for enums

### PR DESCRIPTION
**Pull request by butschster**
| Q             | A
| ------------- | ---
| Bugfix?       | ❌
| Breaks BC?    | ❌
| New feature?  | ✔️

- [x] Unit tests
- [ ] Docs

### Enum 
```php
use Spiral\Boot\EnvironmentInterface;
use Spiral\Boot\Injector\ProvideFrom;
use Spiral\Boot\Injector\InjectableEnumInterface;

#[ProvideFrom(method: 'detect')]
enum AppEnvironment: string implements InjectableEnumInterface
{
    case Production = 'prod';
    case Stage = 'stage';
    case Testing = 'testing';
    case Local = 'local';

    public function isProduction(): bool
    {
        return $this === self::Production;
    }

    public static function detect(EnvironmentInterface $environment): self
    {
        $value = $environment->get('APP_ENV');

        return \is_string($value)
            ? (self::tryFrom($value) ?? self::Local)
            : self::Local;
    }
}
```

### Using

Enum will be automatically resolved by the container with current environment

```php
$appEnv = $container->get(AppEnvironment::class);

dump($appEnv->isProduction());
```

### CODE REVIEW

**Review by msmakouz — APPROVED** [review]

File: src/Boot/src/Environment/DebugMode.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    case Disabled;
+
+    public function isEnabled(): bool
+    {
+        return $this === self::Enabled;
+    }
+
+    public static function detect(EnvironmentInterface $environment): self
+    {
+        return filter_var($environment->get('DEBUG'), FILTER_VALIDATE_BOOL) ? self::Enabled : self::Disabled;
```
Comment (roxblnfk) [inline src/Boot/src/Environment/DebugMode.php:24]:
```suggestion
        return \filter_var($environment->get('DEBUG'), \FILTER_VALIDATE_BOOL) ? self::Enabled : self::Disabled;
```

File: src/Boot/src/Injector/EnumInjector.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public function __construct(
+        private readonly Container $container,
+        private readonly AttributeReader $reader
+    ) {
+    }
+
+    public function createInjection(\ReflectionClass $class, string $context = null): object
+    {
+        $attribute = $this->reader->firstClassMetadata($class, ProvideFrom::class);
+        if (!$attribute) {
```
Comment (roxblnfk) [inline src/Boot/src/Injector/EnumInjector.php:26]:
```suggestion
        if ($attribute === null) {
```

File: src/Boot/src/Injector/EnumInjector.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spiral\Boot\Injector;
+
+use Spiral\Attributes\AttributeReader;
+use Spiral\Core\Container;
+use Spiral\Core\Container\InjectorInterface;
+use Spiral\Core\Exception\Container\ContainerException;
```
Comment (roxblnfk) [inline src/Boot/src/Injector/EnumInjector.php:10]:
```suggestion
use Spiral\Core\Exception\Container\InjectionException;
```

File: src/Boot/src/Injector/EnumInjector.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        private readonly Container $container,
+        private readonly AttributeReader $reader
+    ) {
+    }
+
+    public function createInjection(\ReflectionClass $class, string $context = null): object
+    {
+        $attribute = $this->reader->firstClassMetadata($class, ProvideFrom::class);
+        if (!$attribute) {
+            throw new ContainerException(
```
Comment (roxblnfk) [inline src/Boot/src/Injector/EnumInjector.php:27]:
```suggestion
            throw new InjectionException(
```

File: src/Boot/src/Injector/EnumInjector.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                \sprintf(
+                    'Class `%s` should contain `%s` attribute with defined detector method.',
+                    $class->getName(),
+                    ProvideFrom::class
+                )
+            );
+        }
+
+        if (!$class->isEnum()) {
+            throw new ContainerException(
```
Comment (roxblnfk) [inline src/Boot/src/Injector/EnumInjector.php:37]:
```suggestion
            throw new InjectionException(
```

File: src/Boot/src/Injector/EnumInjector.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                \sprintf(
+                    'Class `%s` should be an enum.',
+                    $class->getName()
+                )
+            );
+        }
+
+        $object = $this->container->invoke(
+            $class->getMethod($attribute->method)->getClosure()
+        );
```
Comment (roxblnfk) [inline src/Boot/src/Injector/EnumInjector.php:47]: Check here please that `$attribute->method` is valid `callable`
Or you can wrap these lines with a `try/catch` block

**Review by roxblnfk — CHANGES_REQUESTED** [review]

**Review by butschster — COMMENTED** [review]
Small fixes

**Review by roxblnfk — APPROVED** [review]

**Pull request merged**